rclcpp/include/rclcpp/parameter_descriptor_wrapper.hpp

```diff
@@ -0,0 +1,102 @@
+#ifndef RCLCPP__PARAMETER_DESCRIPTOR_WRAPPER_HPP_
+#define RCLCPP__PARAMETER_DESCRIPTOR_WRAPPER_HPP_
+
+// Standard library includes
+#include <functional>
+#include <utility>
+#include <memory>
+
+// Additional ROS libraries needed
+#include "node.hpp"
+#include "rcl_interfaces/msg/describe_parameters.hpp"
+#include "rclcpp/node.hpp"
+
+namespace rclcpp
+{
+
+// Implments ParameterDesription class with builder design pattern
+class ParameterDescription
+{
+public:
+    // List of classes the builder manages
+    class FloatingPointDescription;
+
+    // Two constructors in case certain methods don't need to declare_parameters with a Type T in the form of a template
+    ParameterDescription(std::string name, std::uint8_t type, std::string description, std::string additional_constraints, bool read_only. bool dynamic_typing) : m_name{name}, parameter_descriptor.type{type}, m_description{description}, m_additional_constraints{additional_constraints}, read_only{read_only}, dyanmic_typing{m_dyanmic_typing}{};
+
+private:
+    // The main descriptor object we're meant to adjust
+    rcl_interfaces::msg::ParameterDescriptor parameter_descriptor = {};
+
+    // Copies all the information in ParameterDescriptor.msg - https://github.com/ros2/rcl_interfaces/blob/rolling/rcl_interfaces/msg/ParameterDescriptor.msg
```

Comment:
Why do we copy all the information? Shouldn't we just store an internal parameter description message and return it as output of a "build" method?

Reply:
I interpret this as just returning the entirety of the `parameter_descriptor` message that we are currently using to store all the values, which makes the build method appropriate.